ITE/4/README.adoc

```diff
@@ -0,0 +1,283 @@
+as a material of the particular step - it MUST instead be created as part of
+executing this step of the supply chain. This rule is not affected by this ITE.
+
+=== `DELETE <pattern>`
+
+The Delete rule specifies that an artifact matched by this rule MUST not appear
+as a product of the particular step - it MUST instead be deleted as part of
+executing this step of the supply chain. This rule is not affected by this ITE.
+
+=== `REQUIRE <pattern>`
+
+The Require rule specifies that an artifact matched by this rule MUST appear as
+a material or product of the particular step. This rule is not affected by this
+ITE.
+
+=== `MODIFY <pattern>`
+
+The Modify rule specifies that an artifact matched by this rule MUST appear as
+both a material and product of the step, and the hash of the artifact MUST
+differ indicating the execution of the step changed the artifact.
+
+[[motivation]]
+== Motivation
+
+ITE 4 is motivated by the following use cases.
+
+<not just licenses, all spdx materials and "refer" to those instances>
+
+<OCI, Datadog, save space by common references to file hashes>
+
+==== Use Case 1: SPDX Document Identifiers
+
+The Software Package Data Exchange (SPDX) is an open standard for communicating
+software bill of material information such as components, licenses, copyrights,
+and security references. Each document is a comprehensive report that describes
+a software package in detail. They're composed of several entities that have
+unique identifiers. These identifiers can be used in in-toto metadata embedded
+in SPDX documents to refer to the respective entities.
+
+==== Use Case 2: GitHub Entities
+
+GitHub has more abstract entities such as Pull Requests and Issues. These
+entities can be referred to directly using the URI schemes proposed in this ITE
+and help provide attestations about this data.
+
+[[reasoning]]
+== Reasoning
+
+This ITE proposes a change in the URI scheme of artifacts in in-toto metadata.
+The proposed change closely matches the URI structure laid out in RFC 3986. The
+URI structure propsed in that document is widely accepted and is versatile
+enough to allow for a wide variety of references. It is also easy to implement
+and extend support for due to the presence of a large number of standard
+libraries.
+
+[[backwards-compatibility]]
+== Backwards Compatibility
+
+If in-toto metadata is generated using an implementation of in-toto conforming
+to this ITE, verification using a non-conforming implementation will likely
+fail. It is still possible for verification to continue if the inspections don't
+use generic URIs.
+
+However, a conforming implementation should be capable of verifying in-toto
+metadata generated using a non-conforming implementation, as a conforming
+system must also conform to the main in-toto specification.
+
+It's also possible two conforming systems are unable to verify the other's
+in-toto metadata as they're unaware of how to resolve certain URI tokens used by
+the other.
```

Comment:
Isn't this ITE about resolving URIs and hashing the contents of the documents they point to, when **generating** link metadata (e.g. in `in-toto-run`)?

How is verification (e.g. through `in-toto-verify`) affected by this ITE? As you say above, at the time of verification no more URIs are resolved.

The only compatibility issue I can see on the verification side is the link metadata schema. But from quickly looking at the specs and the reference implementation, artifact paths already don't have any restrictions that wouldn't also allow for a "`<scheme>:<hier-part>`"-string.

Reply:
Oh right, inspections. I think it's worth being explicit there.